src/components/layout/sync-button.tsx

```diff
@@ -0,0 +1,161 @@
+import { Button, Box, LinearProgress, Snackbar, Alert } from "@mui/material";
+import { DateTime } from "luxon";
+import { SyntheticEvent, useState, useEffect  } from "react";
+import { useApi } from "../../hooks/use-api";
+import { errorAtom } from "../../atoms/error";
+import { useAtom, useAtomValue, useSetAtom } from "jotai";
+import SyncDialog from "../contexts/sync-handler";
+import strings from "../../localization/strings";
+import { dailyEntriesAtom, personTotalTimeAtom, personsAtom } from "../../atoms/person";
+import { Person } from "../../generated/client";
+import config from "../../app/config";
+import { userProfileAtom } from "../../atoms/auth";
+/**
+ * Sync button component
+ */
+const SyncButton = () => {
+  const { synchronizeApi } = useApi();
+  const setError = useSetAtom(errorAtom);
+  const yesterday: DateTime = DateTime.now().minus({ days: 1 });
+  const [syncStartDate, setSyncStartDate] = useState(yesterday);
+  const [syncing, setSyncing] = useState(false);
+  const [syncSuccess, setSyncSuccess] = useState(false);
+  const [syncHandlerOpen, setSyncHandlerOpen] = useState(false);
+  const [dailyEntries, setDailyEntries] = useAtom(dailyEntriesAtom);
+  const [persons, setPersons] = useAtom(personsAtom);
+  const { personsApi, dailyEntriesApi } = useApi();
+  const userProfile = useAtomValue(userProfileAtom);
+  const personTotalTime = useAtomValue(personTotalTimeAtom);
+
+  useEffect(() => {
+    fetchDailyEntries();
+  },[persons]);
+
+  /**
+   * fetching persons 
+   */
+  const fetchPersons = async () => {
+    if (!persons) {
+      try {
+        const tempPersons = await personsApi.listPersons({});
+        setPersons(tempPersons);
```

Comment:
This updates the atom, it might introduce some extremely weird behaviour on other components, so the persons actually be a state not atom?